src/server/routes/summary/entity/index.js

```diff
@@ -0,0 +1,79 @@
+const _ = require('lodash');
+const { validatorGconvert } = require('../../../external-services/gprofiler/gconvert');
+const { getEntitySummary: getNcbiGeneSummary } = require('../../../external-services/ncbi');
+const { getEntitySummary: getHgncSummary } = require('../../../external-services/hgnc');
+const { getEntitySummary: getUniProtSummary } = require('../../../external-services/uniprot');
+const { DATASOURCES } = require('../../../../models/entity/summary');
+
+/**
+ * entityFetch: Retrieve EntitySummary for a given id from a datasource
+ *
+ * @param { array } localID string(s) that should be fetched
+ * @return { object } value is EntitySummary keyed by the local ID
+ */
+const entityFetch = async ( localIds, dataSource ) => {
+  let eSummary;
+  switch ( dataSource ) {
+    case DATASOURCES.HGNC:
+      eSummary = await getHgncSummary( localIds );
+      break;
+    case DATASOURCES.UNIPROT:
+      eSummary = await getUniProtSummary( localIds );
+      break;
+    default:
+      eSummary = await getNcbiGeneSummary( localIds );
+  }
+  return eSummary;
+};
+
+/**
+ * entitySearch: Search a list of strings for recognized gene/protein identifiers
+ * and return an EntitySummary for each
+ *
+ * @param { array } tokens string(s) that should be queried
+ * @return { object } value is EntitySummary keyed by the input token
+ */
+const entitySearch = async tokens => {
+
+  // look for special tokens
+  if ( tokens.length === 1 ) {
```

Comment:
If tokens is not length 1, then this returns undefined.  Maybe you should throw an error or return {}?

Reply:
In the case of token length >1 it just passes right through, doesn't it?

Reply:
Yeah you are right I didn't notice that there is more to this function.